great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -191,6 +191,62 @@
 T = TypeVar("T", dict, list, str)
 
 
+class DatasourceDict(UserDict):
+    def __init__(self, context: AbstractDataContext):
+        self._context = context
+        self._datasource_store = context._datasource_store
+        self._config_provider = context.config_provider
+        self._fluent_config = context.fluent_config
```

Comment:
Really don't like this but hoping to extract dependencies from the context and inject them so we don't have a cyclic dependency between `DatasourceDict` and `DataContext`